inductiva/_cli/cmd_logs/logs.py

```diff
@@ -1,22 +1,86 @@
 """CLI for logs."""
+import json
+import re
 import sys
+from typing import Tuple
 
 from .. import utils as cli_utils
-from ... import tasks
+from ... import tasks, ApiException
 
 
-def stream_task_logs(args):
-    """Consume the stream logs of a certain task."""
-    task_id = args.task_id
-    task = tasks.Task(task_id)
+def _get_task_id_from_mode(mode: str) -> Tuple[bool, str]:
+    """Extract the task_id from the mode."""
+    match = re.match(r"^(submitted|started)(-(\d+))?$", mode)
+    if not match:
+        return False, f"Invalid mode format: {mode}"
+
+    status_match = match.group(1)
+    offset_match = match.group(3)
+
+    status = status_match
+    offset = (int(offset_match) + 1) if offset_match else 1
+
+    # If the status is submitted, send None to the API to get all the tasks.
+    # The endpoint already returns the tasks in the submitted order by default
+    if status == "submitted":
+        status = None
+
+    task_list = tasks.get(last_n=offset, status=status)
+
+    if not task_list:
+        return False, f"No '{status}' tasks found."
+
+    return True, task_list[-1].id
+
+
+def validate_mode_or_task_id(mode: str) -> Tuple[bool, str]:
+    """Validate the mode or task_id."""
+    if mode.startswith("submitted") or mode.startswith("started"):
```

Comment:
Isn't this checked at `_get_task_id_from_mode` during the regular expression? If the str does not start with submitted or started the regex will fail.

Reply:
Hmmm yes.. they are checked again in the `_get_task_id_from_mode`, the regex checks for the -n parameter as well. Therefore the -n parameter only exists if the `mode` starts with submitted or started. 

If it does not starts with these two strings it wont even enter the `_get_task_id_from_mode` function and it is assumed that the mode is a task_id.